authcrypt/main.cpp

```diff
@@ -24,15 +24,14 @@
 #include "mbedtls/platform.h"
 
 int main() {
-    mbedtls_platform_context platform_ctx;
     int exit_code = MBEDTLS_EXIT_FAILURE;
 
-    if((exit_code = mbedtls_platform_setup(&platform_ctx)) != 0) {
+    if((exit_code = mbedtls_platform_setup(NULL)) != 0) {
```

Comment:
This is assigning an error code from `mbedtls_platform_setup()` to an `exit_code` used to exit `main()`. They are two different sets of values - and the error values don't map to valid exit codes. Initialise `exit_code` to `MBEDTLS_EXIT_SUCCESS` and remove these follow on assignments. They're not needed.

Reply:
@sbutcher-arm I disagree.
Although the exit code is not the return code of the functions, we use it for logging the error code. When we exit the application upon failure, the return code will always be `MBEDTLS_EXIT_FAILURE`

Reply:
Well, I disagree. ;) I think it's reusing the variable for two different purposes with sets of values that don't necessarily mix. But it's a pre-existing problem, so I think I can ignore it for the moment, as the code works, just doesn't match my expectations.

Reply:
Sure, I created #205 to track this